src/test/java/dev/cdevents/CDEventsTest.java

```diff
@@ -42,6 +42,26 @@ void createPipelineRunFinishedEventAsCloudEvent() {
 
     }
 
+    @Test
+    void createPipelineRunFinishedEventOptionalFieldsUnset() {
+
+        PipelineRunFinishedCDEvent cdEvent =  new PipelineRunFinishedCDEvent();
+        cdEvent.setSource(URI.create("http://dev.cdevents"));
+
+        cdEvent.setSubjectId("/dev/pipeline/run/subject");
+
+        String cdEventJson = CDEvents.cdEventAsJson(cdEvent);
+
+        CloudEvent ceEvent = CDEvents.cdEventAsCloudEvent(cdEvent);
+
+        String ceDataJson = new String(ceEvent.getData().toBytes(), StandardCharsets.UTF_8);
+
+        assertThat(ceEvent.getType()).isEqualTo(cdEvent.getContext().getType());
+        assertThat(ceEvent.getSource()).isEqualTo(cdEvent.getContext().getSource());
+        assertThat(ceDataJson).isEqualTo(cdEventJson);
```

Comment:
If I understand the test correctly, what it does it:

- create a CDEvent object, renders it to json
- create a CloudEvent out of the CDEvent
- extract the payload from the CloudEvent and matches is against the CDEvent as json
- compare type and source

This looks good to test check CloudEvent binding, but how does it help to test the unset fields?
Don't we need JSON documents as test data for that?

Reply:
Yes, added test data to compare optional fields with the actual CDEvent.
To keep this PR smaller, added the optional field validation only for Core events for now, will create separate PRs to handle tests for other type of events.